### PR TITLE
[Dynamo] Support SDPAParams construction in fullgraph

### LINKED ISSUE #182684: [`SDPA`] Checking for FA backend eligibility via SDPAParams fails under fullgraph compile
### 🐛 Describe the bug

Currently, the `enable_gqa` using torch sdpa is only viable for the FA and math backend (also probably cuDNN but doesn't really matter here) so we actually only want to use this feature when we know that FA will be invoked as the math backend is too slow compared to the mem-efficient path (with manually repeated kv groups).

We had naive assumptions first, e.g. no mask + viable version. But this is not sufficient, because there are other possible breaking conditions e.g. different seq_lens or head dims. The idea was then to use the native torch checking functions see https://github.com/huggingface/transformers/pull/45776 (because manually keeping track of these will be unmaintainable imo)

However, based on our CI (https://app.circleci.com/pipelines/github/huggingface/transformers/174021/workflows/0632f33d-6844-4576-a08c-04ccbda483de/jobs/2300522 Edit: was on run on CPU so ran into a wrong conditional local cuda GPU failure remains) you can see that a fullgraph test is failing, i.e. `tests/models/glm_ocr/test_modeling_glm_ocr.py::GlmOcrModelTest::test_generate_compile_model_forward_fullgraph`. It seems (to me) that this is related to the `SDPAParams` object not being fully traceable - which honestly shouldn't be the case, no? 

The reproducer hence is to run the aforementioned test on that branch of transformers. I have also verified this locally e.g.
```py
import os

import pytest

os.environ["RUN_SLOW"] = "1"

model = "glm_ocr"
pytest.main([f"tests/models/{model}/test_modeling_{model}.py", "-k", "test_generate_compile_model_forward_fullgraph"])
```

Stacktrace:
```
E               torch._dynamo.exc.Unsupported: Attempted to call function marked as skipped
E                 Explanation: Dynamo does not know how to trace the builtin `<unknown module>.pybind11_object.__new__.` This function is either a Python builtin (e.g. _warnings.warn) or a third-party C/C++ Python extension (perhaps created with pybind).
E                 Hint: If it is a Python builtin, please file an issue on GitHub so the PyTorch team can add support for it and see the next case for a workaround.
E                 Hint: If it is a third-party C/C++ Python extension, please either wrap it into a PyTorch-understood custom operator (see https://pytorch.org/tutorials/advanced/custom_ops_landing_page.html for more details) or, if it is traceable, use `torch.compiler.allow_in_graph`.
E               
E                 Developer debug context: module: <unknown module>, qualname: pybind11_object.__new__, skip reason: <missing reason>
E               
E                For more details about this graph break, please visit: https://meta-pytorch.github.io/compile-graph-break-site/gb/gb0007.html
E               
E               from user code:
E                  File "/home/anton/Code/work/transformers/src/transformers/utils/generic.py", line 900, in wrapper
E                   output = func(self, *args, **kwargs)
E                 File "/home/anton/Code/work/transformers/src/transformers/models/glm_ocr/modeling_glm_ocr.py", line 1368, in forward
E                   outputs = self.model(
E                 File "/home/anton/Code/work/transformers/src/transformers/utils/generic.py", line 900, in wrapper
E                   output = func(self, *args, **kwargs)
E                 File "/home/anton/Code/work/transformers/src/transformers/models/glm_ocr/modeling_glm_ocr.py", line 1214, in forward
E                   outputs = self.language_model(
E                 File "/home/anton/Code/work/transformers/src/transformers/utils/generic.py", line 976, in wrapper
E                   output = func(self, *args, **kwargs)
E                 File "/home/anton/Code/work/transformers/src/transformers/utils/output_capturing.py", line 248, in wrapper
E                   outputs = func(self, *args, **kwargs)
E                 File "/home/anton/Code/work/transformers/src/transformers/models/glm_ocr/modeling_glm_ocr.py", line 825, in forward
E                   layer_outputs = decoder_layer(
E                 File "/home/anton/Code/work/transformers/src/transformers/modeling_layers.py", line 93, in __call__
E                   return super().__call__(*args, **kwargs)
E                 File "/home/anton/Code/work/transformers/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1779, in _wrapped_call_impl
E                   return self._call_impl(*args, **kwargs)
E                 File "/home/anton/Code/work/transformers/.venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1790, in _call_impl
E                   return forward_call(*args, **kwargs)
E                 File "/home/anton/Code/work/transformers/src/transformers/models/glm_ocr/modeling_glm_ocr.py", line 282, in forward
E                   hidden_states, _ = self.self_attn(
E                 File "/home/anton/Code/work/transformers/src/transformers/models/glm_ocr/modeling_glm_ocr.py", line 221, in forward
E                   attn_output, attn_weights = attention_interface(
E                 File "/home/anton/Code/work/transformers/src/transformers/integrations/sdpa_attention.py", line 110, in sdpa_attention_forward
E                   if not use_gqa_in_sdpa(
E                 File "/home/anton/Code/work/transformers/src/transformers/integrations/sdpa_attention.py", line 59, in use_gqa_in_sdpa
E                   params=SDPAParams(query, key, value, attention_mask, dropout, is_causal, True)
E                 File "/home/anton/Code/work/transformers/.venv/lib/python3.12/site-packages/torch/_dynamo/polyfills/__init__.py", line 405, in instantiate_user_defined_class_object
E                   obj = cls.__new__(cls, *args, **kwargs)
E               
E               Set TORCHDYNAMO_VERBOSE=1 for the internal stack trace (please do this especially if you're reporting a bug to PyTorch). For even more developer context, set TORCH_LOGS="+dynamo"

.venv/lib/python3.12/site-packages/torch/_dynamo/eval_frame.py:1034: Unsupported
```

### Versions

This is my local install since I don't think the HW matters too much in this instance

PyTorch version: 2.11.0+cu128
Is debug build: False
CUDA used to build PyTorch: 12.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.4 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.39

Python version: 3.12.3 (main, Mar 23 2026, 19:04:32) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-6.17.0-115022-tuxedo-x86_64-with-glibc2.39
Is CUDA available: True
CUDA runtime version: 12.0.140
CUDA_MODULE_LOADING set to: 
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 4060 Laptop GPU
Nvidia driver version: 580.126.09
cuDNN version: Could not collect
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           39 bits physical, 48 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  32
On-line CPU(s) list:                     0-31
Vendor ID:                               GenuineIntel
Model name:                              Intel(R) Core(TM) i9-14900HX
CPU family:                              6
Model:                                   183
Thread(s) per core:                      2
Core(s) per socket:                      24
Socket(s):                               1
Stepping:                                1
CPU(s) scaling MHz:                      17%
CPU max MHz:                             5800.0000
CPU min MHz:                             800.0000
BogoMIPS:                                4838.40
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid rdseed adx smap clflushopt clwb intel_pt sha_ni xsaveopt xsavec xgetbv1 xsaves split_lock_detect user_shstk avx_vnni dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp hwp_pkg_req hfi vnmi umip pku ospke waitpkg gfni vaes vpclmulqdq rdpid movdiri movdir64b fsrm md_clear serialize arch_lbr ibt flush_l1d arch_capabilities
Virtualization:                          VT-x
L1d cache:                               896 KiB (24 instances)
L1i cache:                               1.3 MiB (24 instances)
L2 cache:                                32 MiB (12 instances)
L3 cache:                                36 MiB (1 instance)
NUMA node(s):                            1
NUMA node0 CPU(s):                       0-31
Vulnerability Gather data sampling:      Not affected
Vulnerability Ghostwrite:                Not affected
Vulnerability Indirect target selection: Not affected
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Not affected
Vulnerability Old microcode:             Not affected
Vulnerability Reg file data sampling:    Mitigation; Clear Register File
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Not affected
Vulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Not affected
Vulnerability Tsx async abort:           Not affected
Vulnerability Vmscape:                   Mitigation; IBPB before exit to userspace

Versions of relevant libraries:
[pip3] Could not collect
[conda] Could not collect

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @amjames @Lucaskabela @jataylo @azahed98 @drisspg @liangel-02 @howardzhang-cv

### PR DESCRIPTION
**Pull request by guan404ming**
Fixes #182684

## Why

- Constructing `SDPAParams` inside a compiled region graph-breaks under `fullgraph=True`: the pybind11 class is wrapped as a user-defined class, even though a construction handler already exists

## How

- Routes `torch.backends.cuda.SDPAParams` to `TorchInGraphFunctionVariable`, following the `torch.nn.Parameter` pattern
- Drops the `allow_in_graph` workaround in `test/dynamo/test_sdpa.py`, which is no longer needed

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @jataylo @azahed98

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Review by williamwen42 — APPROVED** [review]

**Comment by guan404ming** [comment]
Thanks!
